rfcs/text/0033/io.yml

```diff
@@ -0,0 +1,48 @@
+- name: io
+  title: IO
+  group: 2
+  short: Fields describing a batch of input or output data produced by a process.
+  description: >
+    Fields describing a batch of input or output (IO) data produced by a process.
+    These fields should only appear on the top level process object when event.action is 'io'
+  type: group
+  reusable:
+    top_level: false
+    expected:
+      - process
+
+  fields:
+
+    - name: fd
+      level: extended
+      type: number
+      description: File descriptor used. 1 = stdout, 2 = stderr
+
+    - name: data
+      level: extended
+      type: object
+      short: An array of timestamped raw input/output lines encoded as UTF-8.
+      description: >
+        An array of timestamped raw input/output lines encoded as UTF-8 characters.
+
+        Each event can contain up to <batch size> (configurable) lines of data. With a maximum line length of <X> (needs discussion, regex?, scrubbing?).
+
+      normalize: array
+
+    - name: data.time
+      level: extended
+      type: date
+      short: Timestamp for the line of IO
+      description: >
+        Timestamp for the line of IO
+
+        Timestamps are tracked per line, so that they can be replayed in realtime using libraries like xtermjs.org.
+
+    - name: data.value
```

Comment:
Would it be better to be more explicit in the 'value' field.
Would there be any need in the future to separate this out from '.raw' to '.something_else'?

Reply:
Is this in regards to potentially storing the same info as plain text (w/o escape codes)? It's still not apparent how much of an issue this might be (formatting between words etc..), but if so, one could imagine a schema that has both:
process.io.data.value (or process.io.data.raw)
and process.io.data.text